fv3gfs/physics/stencils/fv_update_phys.py

```diff
@@ -0,0 +1,212 @@
+    the FV3 discretization and definition of the prognostic variables
+    """
+
+    def __init__(
+        self, grid, namelist, comm: fv3gfs.util.CubedSphereCommunicator,
+    ):
+        self.grid = grid
+        self.namelist = namelist
+        self.comm = comm
+        self._dt = Float(self.namelist.dt_atmos)
+        self._moist_cv = FrozenStencil(
+            moist_cv,
+            origin=self.grid.compute_origin(),
+            domain=self.grid.grid_indexing.domain_compute(add=(0, 0, 1)),
+        )
+        self._update_pressure_and_surface_winds = FrozenStencil(
+            update_pressure_and_surface_winds,
+            origin=self.grid.compute_origin(),
+            domain=self.grid.grid_indexing.domain_compute(add=(0, 0, 1)),
+        )
+        self._AGrid2DGridPhysics = AGrid2DGridPhysics(self.grid, self.namelist)
+        self._do_cubed_to_latlon = CubedToLatLon(
+            self.grid.grid_indexing, self.grid.grid_data, order=namelist.c2l_ord
+        )
+        origin = self.grid.grid_indexing.origin_compute()
+        shape = self.grid.grid_indexing.max_shape
+        full_3Dfield_1pts_halo_spec = self.grid.grid_indexing.get_quantity_halo_spec(
+            shape, origin, dims=[fv3util.X_DIM, fv3util.Y_DIM, fv3util.Z_DIM], n_halo=1,
+        )
+        self._udt_halo_updater = self.comm.get_scalar_halo_updater(
+            [full_3Dfield_1pts_halo_spec]
+        )
+        self._vdt_halo_updater = self.comm.get_scalar_halo_updater(
+            [full_3Dfield_1pts_halo_spec]
+        )
+        # TODO: check if we actually need surface winds
+        self._u_srf = utils.make_storage_from_shape(
+            shape[0:2], origin=origin, init=True
+        )
+        self._v_srf = utils.make_storage_from_shape(
+            shape[0:2], origin=origin, init=True
+        )
+
+    def __call__(
+        self,
+        state,
+        u_dt: FloatField,
+        v_dt: FloatField,
+        t_dt: FloatField,
+        # The following should be generated by grid initialization
+        vlon1: FloatFieldIJ,
+        vlon2: FloatFieldIJ,
+        vlon3: FloatFieldIJ,
+        vlat1: FloatFieldIJ,
+        vlat2: FloatFieldIJ,
+        vlat3: FloatFieldIJ,
+        edge_vect_w: FloatFieldIJ,
+        edge_vect_e: FloatFieldIJ,
+        edge_vect_s: FloatFieldI,
+        edge_vect_n: FloatFieldI,
+        es1_1: FloatFieldIJ,
+        es2_1: FloatFieldIJ,
+        es3_1: FloatFieldIJ,
+        ew1_2: FloatFieldIJ,
+        ew2_2: FloatFieldIJ,
+        ew3_2: FloatFieldIJ,
+    ):
+        self._moist_cv(
+            state.qvapor,
+            state.qliquid,
+            state.qrain,
+            state.qsnow,
+            state.qice,
+            state.qgraupel,
+            state.pt,
+            t_dt,
+            cp_air,
+            self._dt,
+        )
+        # [TODO] needs a better solution to handle u_dt, v_dt quantity
+        u_dt_quantity = self.grid.make_quantity(u_dt)
```

Comment:
Move those allocation in __init__ with the [TODO]

Reply:
The problem is that at init time we don't have the variables. I think the proper way to do this is to have something like the dycore, where each variable has storage and `_quantity`. The thing is that the only variables in physics that need to be quantity is these two (`u_dt`, `v_dt`). Does that still make sense to have quantity for every variable?

Reply:
Quantity doesn't exist only to provide halo updates, but also to provide documentation on dims and units. It's really just metadata attached to a Storage. I wouldn't worry too much about using Quantity for every variable if it solves a problem you're having.